### ISSUE TITLE
--verifytypes counts re-exported symbols twice (or more)

### ISSUE THREAD

**Issue by matangover** [microsoft/pyright#4009]
**Describe the bug**
`pyright --verifytypes` counts any re-exported symbol in both the original module where it's defined and the module which re-exports it. For packages that use a lot of `import *`, this can mean thousands of duplicates.

**To Reproduce**
Create a python package with the following structure:
```python
# typing_test/__init__.py
from .module import *
# typing_test/module.py
def test():
    return 3
# typing_test/py.typed
<empty file>
```
Install the package and run `pyright --verify-types typing_test`. The result should show ```Symbols exported [...] with unknown type: 2```

**Expected behavior**
Each symbol should be counted only once, in one of its public names (doesn't really matter which one). Or this should at least be an option.

**VS Code extension or command-line**
Command line pyright 1.1.274

**Comment by erictraut** [comment]
The current behavior is by design. 

There are two different public symbols in your example above: `typing_test/test` and `typing_test/module/test`. These are two distinct symbols even though they happen to share the same type. Both of these symbols are public, so they are both counted as part of the `--verifytypes` analysis.

Note that only public symbols are counted and reported. If you rename the `module.py` in your example above to `_module.py`, only the `typing_test/test` symbol will be considered public and flagged as incomplete.

**Comment by matangover** [comment]
Let me expand. I am adding type annotations to librosa. It uses a lot of `import *`. I honestly don't care about the reported percentage too much, I just care about the list of errors that is printed which helps me improve the annotations. Right now every error is printed several times and it makes it very difficult for me to decide what errors I should focus on. Thus, an option of `--exclude-reexported` would be very useful :)

As a workaround, I added `if not TYPE_CHECKING` around the `import *` statements.

**Comment by erictraut** [comment]
Another solution is to use the `--outputjson` command line option, which will emit a JSON report. You can then easily write a script that filters or transforms the output any way you like.

**Comment by matangover** [comment]
Oh, nice :)
